cpp/src/mip/problem/problem.cu

```diff
@@ -1039,13 +1049,79 @@ void problem_t<i_t, f_t>::fix_given_variables(problem_t<i_t, f_t>& original_prob
                                               const rmm::device_uvector<i_t>& variables_to_fix,
                                               const raft::handle_t* handle_ptr)
 {
-  i_t TPB = 64;
-  fix_given_variables_kernel<i_t, f_t><<<n_constraints, TPB, 0, handle_ptr->get_stream()>>>(
-    original_problem.view(),
-    view(),
-    raft::device_span<f_t>{assignment.data(), assignment.size()},
-    raft::device_span<i_t>{const_cast<i_t*>(variables_to_fix.data()), variables_to_fix.size()});
+  rmm::device_uvector<f_t> reduction_in_rhs(n_constraints, handle_ptr->get_stream());
```

Comment:
Should we move the allocations in a struct? If it is called multiple times with large sizes the overhead could be non-negligible.